pkg/config/apiserver.go

```diff
@@ -55,3 +65,60 @@ type AuditLog struct {
 	// +kubebuilder:default=Default
 	Profile string `json:"profile"`
 }
+
+type TLSConfig struct {
+	// CipherSuites lists the allowed cipher suites that the API server will
+	// accept and serve. Defaults to cipher suites from the minVersion config
+	// parameter.
+	CipherSuites []string `json:"cipherSuites"`
+
+	// MinVersion specifies which TLS version is the minimum version of TLS
+	// to serve from the API server. Allowed values: VersionTLS12, VersionTLS13.
+	// Defaults to VersionTLS12.
+	// +kubebuilder:validation:Enum:=VersionTLS12;VersionTLS13
+	// +kubebuilder:default=VersionTLS12
+	MinVersion string `json:"minVersion"`
+}
+
+func (t *TLSConfig) UpdateValues() {
+	if t.MinVersion == "" {
+		t.MinVersion = string(configv1.VersionTLS12)
+	}
+	switch t.MinVersion {
+	case string(configv1.VersionTLS12):
+		if len(t.CipherSuites) == 0 {
+			t.CipherSuites = getIANACipherSuites(configv1.TLSProfiles[configv1.TLSProfileIntermediateType].Ciphers)
+		}
+	case string(configv1.VersionTLS13):
+		// Golang does not allow specifying cipher suites when using tls 1.3, so we
+		// override whatever the user has configured to match the cipher suites that
+		// will actually be used. Note that this is only for informational purposes,
+		// Golang will ignore whatever is configured.
+		t.CipherSuites = getIANACipherSuites(configv1.TLSProfiles[configv1.TLSProfileModernType].Ciphers)
+	}
+}
+
+func (t *TLSConfig) Validate() error {
+	if len(t.CipherSuites) == 0 {
+		return fmt.Errorf("unsupported empty cipher suites")
+	}
+	var cipherSuites []string
```

Comment:
This might make it more obvious
```suggestion
	var allowedCipherSuites []string
```

Reply:
Was keeping the syntax from other control plane components, such as kubelet or apiserver. All of them use `--tls-cipher-suites` and `-tls-min-version`. The ingress operator [also uses](https://github.com/openshift/cluster-ingress-operator/blob/8be174945db79b0660bd0fd21ed279ab7b6aafbf/manifests/02-deployment.yaml#L98-L100) this syntax. Should we change it to make it different?